baseapp/auth/saml/params.go

```diff
@@ -202,3 +202,13 @@ func WithNameIDFormat(n saml.NameIDFormat) Param {
 		return nil
 	}
 }
+
+// WithEncryptedAssertions enables or disables assertion encryption. By
+// default, encryption is enabled. When set to false, the encryption key is not
+// included in generated metadata.
+func WithEncryptedAssertions(encrypt bool) Param {
+	return func(sp *ServiceProvider) error {
+		sp.disableEncryption = !encrypt
```

Comment:
Having the internal option be "disable" is best because the zero value is the correct default and it describes when we need to take action, but it felt like using the reverse was better for the public option. I'm open to changing either side if anyone thinks that is better.